src/Corax/IndexWriter.cs

```diff
@@ -29,14 +30,106 @@ public enum TermIdMask : long
         Set = 2
     }
 
+
+    public class IndexFieldsMapping : IEnumerable<IndexFieldBinding>
+    {
+        public static readonly IndexFieldsMapping Instance = new IndexFieldsMapping(null);
+
+        private readonly ByteStringContext _context;
+        private readonly Dictionary<Slice, IndexFieldBinding> _fields;
+        private readonly Dictionary<int, IndexFieldBinding> _fieldsById;
+        private readonly List<IndexFieldBinding> _fieldsList;
+
+        public int Count => _fieldsById.Count;
+
+        public IndexFieldsMapping(ByteStringContext context)        
+        {
+            _context = context;
+            _fields = new Dictionary<Slice, IndexFieldBinding>();
+            _fieldsById = new Dictionary<int, IndexFieldBinding>();
+            _fieldsList = new List<IndexFieldBinding>();
+        }
+
+        public IndexFieldsMapping AddBinding(int fieldId, Slice fieldName, Analyzer analyzer = null)
```

Comment:
Why do we need to provide `fieldId` externally? Shouldn't we just use `_fields.Count+1` ?

Reply:
I've modified `IndexDefiniton` to have also `fieldId`. This way we've confidence that ids are always identical.

Reply:
It is important for Corax to have `fieldId` if RavenDB then decides to use sequential and do not store it is fine. But the internal of Corax uses those `fieldId` values explicitely.